src/xml_document.cc

```diff
@@ -335,144 +350,112 @@ xmlParserOption getParserOptions(v8::Local<v8::Object> props) {
+    char* baseUrl = getStringOption(
+                options->Get(Nan::New<v8::String>("baseUrl").ToLocalChecked()));
+
+    char* encoding = getStringOption(
+                options->Get(Nan::New<v8::String>("encoding").ToLocalChecked()));
+
+    int parserOptions = getParserOptions(options);
 
-    v8::Local<v8::Object> options = info[1]->ToObject();
-    v8::Local<v8::Value>  baseUrlOpt  = options->Get(
-        Nan::New<v8::String>("baseUrl").ToLocalChecked());
-    v8::Local<v8::Value>  encodingOpt = options->Get(
-        Nan::New<v8::String>("encoding").ToLocalChecked());
-    v8::Local<v8::Value> excludeImpliedElementsOpt = options->Get(
-        Nan::New<v8::String>("excludeImpliedElements").ToLocalChecked());
-
-    // the base URL that will be used for this HTML parsed document
-    v8::String::Utf8Value baseUrl_(baseUrlOpt->ToString());
-    const char * baseUrl = *baseUrl_;
-    if (!baseUrlOpt->IsString()) {
-        baseUrl = NULL;
+    if (node::Buffer::HasInstance(data)) {
+        v8::Local<v8::Object> buf = data->ToObject();
+        return parseString(node::Buffer::Data(buf),
+                            node::Buffer::Length(buf),
+                            baseUrl,
+                            encoding,
+                            parserOptions,
+                            isHTML);
+    }
+    else {
+        v8::String::Utf8Value str(data->ToString());
+        return parseString(*str,
+                            str.length(),
+                            baseUrl,
+                            encoding,
+                            parserOptions,
+                            isHTML);
     }
 
-    // the encoding to be used for this document
-    // (leave NULL for libxml to autodetect)
-    v8::String::Utf8Value encoding_(encodingOpt->ToString());
-    const char * encoding = *encoding_;
 
-    if (!encodingOpt->IsString()) {
-        encoding = NULL;
+    /*
+    xmlNode* root_node = xmlDocGetRootElement(doc);
+    if (root_node == NULL) {
+        return Nan::ThrowError("parsed document has no root element");
     }
+    */
+}
+
+NAN_METHOD(XmlDocument::FromHtml)
+{
+    Nan::HandleScope scope;
 
     v8::Local<v8::Array> errors = Nan::New<v8::Array>();
     xmlResetLastError();
     xmlSetStructuredErrorFunc(reinterpret_cast<void*>(&errors), XmlSyntaxError::PushToArray);
 
-    int opts = (int)getParserOptions(options);
-    if (excludeImpliedElementsOpt->ToBoolean()->Value())
-        opts |= HTML_PARSE_NOIMPLIED | HTML_PARSE_NODEFDTD;
-
-    htmlDocPtr doc;
-    if (!node::Buffer::HasInstance(info[0])) {
-        // Parse a string
-        v8::String::Utf8Value str(info[0]->ToString());
-        doc = htmlReadMemory(*str, str.length(), baseUrl, encoding, opts);
-    }
-    else {
-        // Parse a buffer
-        v8::Local<v8::Object> buf = info[0]->ToObject();
-        doc = htmlReadMemory(node::Buffer::Data(buf), node::Buffer::Length(buf),
```

Comment:
These are changes I brought in to see if it resolved (or caused more for that matter) any build issues in Windows. I think it'd better suited in a separate PR where it can be reviewed on its own. I'm going to updated this PR and revert xml_document back to the master branch version.